coin_change_problem/CoinChange.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Implementation of famous dynamic programming problem
+ *  that aims to find out the maximum number of ways in
+ *  which a value can be achieved using some fixed valued
+ *  coins.
+ *
+ * In the implementation, the time complexity is O(mn)
+ *  and extra space required is O(n).
+ */
+import java.util.Arrays;
+
+/*
+ * coin: Array containing value of coins
+ * m: Number of types of coins
+ * n: Value to find the change for
+ */
+public class CoinChange 
+{
+    static int coin_change_problem(int C[], int m, int n)

```

Comment:
Square brackets in java should be with its type descriptor as `int[] C`. Also follow `camelCase` for naming methods.